platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -339,28 +339,13 @@ def get_change_event(self, timeout=0):
             status = self.sfp_event.check_sfp_status(port_dict, error_dict, timeout)
 
         if status:
-            self.reinit_sfps(port_dict)
             result_dict = {'sfp':port_dict}
             if error_dict:
                 result_dict['sfp_error'] = error_dict
             return True, result_dict
         else:
             return True, {'sfp':{}}
 
-    def reinit_sfps(self, port_dict):
```

Comment:
this function was added to address an issue in a corner case: replace a QSFP module with an SFP module(with a QSA adaptor) in the fly, w/o reinit, sfp object will not update the module type thus not able to identify the new type of module correctly, not sure whether the new implementation can overcome this case w/o this function.

Reply:
Thanks for letting me know. Now I understand the purpose of reinit method. Updated the implementation accordingly.